modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQSession.java

```diff
@@ -2040,6 +2040,10 @@ protected C createConsumerImpl(final Destination destination, final int prefetch
                                                  final int prefetchLow, final boolean noLocal, final boolean exclusive, String selector, final FieldTable rawSelector,
                                                  final boolean noConsume, final boolean autoClose) throws JMSException
     {
+        if (this._consumers.values().size() > 0 ) {
+            throw new JMSException("Only one consumer can be created per session");
```

Comment:
We cannot support having multiple consumers in a single session with the current architecture. Will try to update the server code.